core/src/main/java/io/questdb/griffin/engine/functions/window/RowNumberFunctionFactory.java

```diff
@@ -144,16 +144,20 @@ public int getPassCount() {
             return WindowFunction.ZERO_PASS;
         }
 
-        @Override
-        public void initRecordComparator(RecordComparatorCompiler recordComparatorCompiler, ArrayColumnTypes chainTypes, IntList order) {
-        }
-
         @Override
         public void pass1(Record record, long recordOffset, WindowSPI spi) {
             computeNext(record);
             Unsafe.getUnsafe().putLong(spi.getAddress(recordOffset, columnIndex), rowNumber);
         }
 
+        @Override
+        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+            super.init(symbolTableSource, executionContext);
+            if (partitionByRecord != null) {
```

Comment:
`partitionByRecord` seems to be non-null in this class, so this check is redundant. Might be a good idea to add `@NotNull` annotation to the arguments in the function constructor to stress that.